### DISCUSSION TITLE
Default keys and aliases location on non-Unix platforms

### DISCUSSION THREAD

**Discussion by str4d** [FiloSottile/age#254] [Spec feedback]
The specification gives the Unix path `~/.config/age/*` as the default location it looks for keys and aliases. How should this map to Windows and macOS? The [directories](https://crates.io/crates/directories) Rust crate (which aims to follow per-OS conventions) appears to indicate the following as being approximately OS-shaped:

- Windows: `C:\Users\Alice\AppData\Roaming\age tool\age\config\*`
- macOS: `/Users/Alice/Library/Preferences/com.age-tool.age/*`

We could also go simpler by just using the known folders for program settings as the base:

- Windows: `C:\Users\Alice\AppData\Roaming\age\*`
- macOS: `/Users/Alice/Library/Preferences/age/*`

Thoughts? Is there somewhere else these should go? Are users expected to be editing these by hand (in which case maybe a Documents-like folder should be used, which is more accessible)?

**Comment by danlamanna** [comment]
FWIW, It looks like the Go team had a lot of [discussion](https://github.com/golang/go/issues/29960) on this topic for the 1.13 release and produced [os.UserConfigDir](https://golang.org/pkg/os/#UserConfigDir). Perhaps `age` should rely on that?

**Comment by str4d** [comment]
I scanned the discussion there and it seems like a reasonable approach. For my Rust impl to be compatible I'd need to manually detect macOS and handle it separately because of how the `directories` crate differs, but that's not too bad.

**Comment by Fastidious** [comment]
I think `age` should use `$home/.config/age/` under macOS as well. Several applications share the same directory.

**Comment by FiloSottile** [comment]
For the record, we ended up removing the `~/.config/age` path entirely, and replacing it with a recommendation that applications generate per-task keys and keep them in the application config path.
